tendrl/gluster_integration/sds_sync/__init__.py

```diff
@@ -930,3 +924,91 @@ def get_volume_alert_counts():
                                          'alert_count': 0
                                          }
     return alert_counts
+
+
+def get_device_tree():
+    devicetree = {}
+    try:
+        columns = 'NAME,KNAME,PKNAME,TYPE,MOUNTPOINT,SIZE'
+        keys = columns.split(',')
+        lsblk = ("lsblk --all --bytes --noheadings "
+                 "--output='%s' --path --raw" % columns)
+        cmd = cmd_utils.Command(lsblk)
+        out, err, rc = cmd.run()
+        if not err:
+            out = out.encode('utf8')
+            devlist = map(
+                lambda line: dict(zip(keys, line.split(' '))),
+                out.splitlines()
+            )
+            block_devices = {}
+            for dev_info in devlist:
+                if dev_info["TYPE"] == "disk":
+                    # create new block device list
+                    block_devices = {dev_info["KNAME"]: dev_info}
+                else:
+                    block_devices[dev_info["KNAME"]] = dev_info
+                # find ancestors if mountpoint present
+                if dev_info["MOUNTPOINT"]:
+                    ancestor_details = find_ancestors(
+                        block_devices, dev_info["KNAME"]
+                    )
+                    if dev_info["MOUNTPOINT"] in devicetree.keys():
+                        # if different block device mounted in same mountpoint
+                        # then append the disk and partition information under
+                        # same mountpoint
+                        disks = devicetree[dev_info["MOUNTPOINT"]]["disks"]
+                        devicetree[dev_info["MOUNTPOINT"]]["disks"] = list(
+                            set(disks + ancestor_details["disks"])
+                        )
+                        devicetree[dev_info["MOUNTPOINT"]]["partitions"] = \
+                            list(
+                                set(ancestor_details["partitions"])
+                            )
+                    else:
+                        devicetree[dev_info["MOUNTPOINT"]] = {}
+                        devicetree[dev_info["MOUNTPOINT"]]["disks"] = \
+                            ancestor_details["disks"]
+                        devicetree[dev_info["MOUNTPOINT"]]["partitions"] = \
+                            ancestor_details["partitions"]
+                        devicetree[dev_info["MOUNTPOINT"]]["device_info"] = {}
+                        devicetree[dev_info["MOUNTPOINT"]]["device_info"][
+                            "size"] = dev_info["SIZE"]
+                        devicetree[dev_info["MOUNTPOINT"]]["device_info"][
+                            "mountpoint"] = dev_info["MOUNTPOINT"]
+                        devicetree[dev_info["MOUNTPOINT"]]["device_info"][
+                            "type"] = dev_info["TYPE"]
+                        devicetree[dev_info["MOUNTPOINT"]]["device_info"][
+                            "size"] = dev_info["SIZE"]
+        return devicetree
+    except Exception as ex:
```

Comment:
What kind of exception we expect here? Why not catch the specific ones and take action?

Reply:
i will replace this with specific exception